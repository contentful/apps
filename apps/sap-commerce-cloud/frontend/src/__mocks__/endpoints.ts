export const mockApiEndpoint = 'localhost:9002';
export const mockBaseSite = 'electronics-spa';
