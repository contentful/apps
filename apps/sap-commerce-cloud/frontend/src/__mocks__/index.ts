export { makeSdkMock } from './mockSdk';
export { mockProductPreview } from './mockProductPreview';
