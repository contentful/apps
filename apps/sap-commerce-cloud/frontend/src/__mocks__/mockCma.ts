const mockCma: any = {
  appSignedRequest: {
    create: () => ({}),
  },
  getSpace: () => ({
    getEnvironment: () => ({
      getEditorInterfaces: () => ({
        items: [],
      }),
      getContentTypes: () => ({
        items: [],
      }),
    }),
  }),
};

export { mockCma };
