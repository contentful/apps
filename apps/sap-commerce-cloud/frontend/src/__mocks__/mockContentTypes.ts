export const mockContentTypes = [
  {
    sys: { id: 'ct1' },
    name: 'ct1',
    fields: [
      { id: 'f1', name: 'f1', type: 'Symbol' },
      { id: 'f2', name: 'f2', type: 'Array', items: { type: 'Symbol' } },
      { id: 'f3', name: 'f3', type: 'Array', items: { type: 'Link' } },
      { id: 'f4', name: 'f4', type: 'Link' },
    ],
  },
  {
    sys: { id: 'ct2' },
    name: 'ct2',
    fields: [
      { id: 'f5', name: 'f5', type: 'Symbol' },
      { id: 'f6', name: 'f6', type: 'Array', items: { type: 'Symbol' } },
    ],
  },
];
