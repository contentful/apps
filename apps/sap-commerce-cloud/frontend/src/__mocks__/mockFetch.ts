import { vi } from 'vitest';

export const mockFetch = (response: any) =>
  vi.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(response),
      ok: true,
      status: 200,
      statusText: 'OK',
      headers: new Headers(),
      redirected: false,
      type: 'basic' as ResponseType,
      url: '',
      clone: () => new Response(),
      body: null,
      bodyUsed: false,
      arrayBuffer: () => Promise.resolve(new ArrayBuffer(0)),
      blob: () => Promise.resolve(new Blob()),
      formData: () => Promise.resolve(new FormData()),
      text: () => Promise.resolve(''),
    })
  );
