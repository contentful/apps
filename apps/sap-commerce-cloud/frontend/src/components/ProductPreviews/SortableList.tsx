import { DeleteFn, Product } from '@interfaces';
import { SortableListItem } from '@components/ProductPreviews/SortableListItem';
import { styles } from '@components/ProductPreviews/SortableList.styles';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
}

export function SortableList({ disabled, deleteFn, productPreviews = [] }: Props) {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div className={styles.container}>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
          />
        );
      })}
    </div>
  );
}
