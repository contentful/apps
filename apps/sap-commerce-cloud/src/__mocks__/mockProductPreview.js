export const mockProductPreview = {
  sku: 'abc1234',
  image: '',
  id: '123',
  name: 'Mock Category',
  isMissing: false,
  productUrl: '/product/abc1234',
};
