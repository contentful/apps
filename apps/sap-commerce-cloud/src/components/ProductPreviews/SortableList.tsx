import { FC } from 'react';
import { DeleteFn, Product } from '../../interfaces';
import { SortableListItem } from './SortableListItem';
import { styles } from './SortableList.styles';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
}

export const SortableList: FC<Props> = ({ disabled, deleteFn, productPreviews = [] }) => {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div className={styles.container}>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
          />
        );
      })}
    </div>
  );
};
