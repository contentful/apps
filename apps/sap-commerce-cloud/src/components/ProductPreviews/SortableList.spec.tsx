import { render, cleanup } from '@testing-library/react';
import { Props, SortableList } from './SortableList';
import { mockProductPreview } from '../../__mocks__';

const defaultProps: Props = {
  disabled: false,
  productPreviews: [mockProductPreview],
  deleteFn: jest.fn(),
};

const renderComponent = (props: Props) => {
  return render(<SortableList {...props} />);
};

describe('SortableList', () => {
  afterEach(cleanup);

  it('should render successfully', async () => {
    const { queryAllByTestId } = renderComponent(defaultProps);

    expect(queryAllByTestId(`sortable-list-item`)).toHaveLength(1);
  });
});
