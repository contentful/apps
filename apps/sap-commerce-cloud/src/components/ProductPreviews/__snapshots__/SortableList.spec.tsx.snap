// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableList should render successfully 1`] = `
<div>
  <div>
    <div
      class="Card__Card___1_26G css-pivbrd Card__Card--padding-default___xgDF9"
      data-test-id="cf-ui-card"
    >
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu css-eiov7n"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-0)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient-1);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-0"
          >
            <rect
              data-test-id="cf-ui-skeleton-image"
              height="48"
              rx="0"
              ry="0"
              width="48"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient-1"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation-2"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-2.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-2.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
      <div
        class="css-kx63wx"
      >
        <img
          alt="Mock Category"
          data-test-id="image"
          src=""
          style="display: none;"
        />
      </div>
      <section
        class="css-1phcpb1"
      >
        <div
          class="Typography__Typography___1ZUfE"
          data-test-id="cf-ui-text-container"
        >
          <h1
            class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
            data-test-id="cf-ui-heading"
          >
            Mock Category
          </h1>
          <h2
            class="Subheading__Subheading___2mA9j css-ksy8vh f36-margin-bottom--m"
            data-test-id="cf-ui-subheading"
          >
            abc1234
          </h2>
        </div>
      </section>
      <div
        class="css-mfolfb"
      >
        <button
          class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
            tabindex="-1"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--primary___3M4Il IconButton__IconButton__icon___3yZQN"
              data-test-id="cf-ui-icon"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
              />
            </svg>
            <span
              class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
